dali/test/python/test_operator_multipaste.py

```diff
@@ -0,0 +1,216 @@
+    for i in range(batch_size):
+        for j in range(iters):
+            while True:
+                in_idx = np.int32(np.random.randint(batch_size))
+                out_idx = np.int32(i if even_paste_count else np.random.randint(batch_size))
+                shape = [np.int32(
+                    np.random.randint(
+                        min(input_size[i], output_size[i]) // (iters if no_intersections else 1)
+                    ) + 1
+                ) for i in range(dim)] if not full_input else input_size
+
+                in_anchor = [np.int32(np.random.randint(input_size[i] - shape[i] + 1)) for i in range(dim)] \
+                    if not in_anchor_top_left else [0] * dim
+
+                out_anchor = [np.int32(np.random.randint(output_size[i] - shape[i] + 1)) for i in range(dim)] \
+                    if not out_anchor_top_left else [0] * dim
+
+                if no_intersections:
+                    is_ok = True
+                    for k in range(len(in_idx_l[out_idx])):
+                        if intersects(out_anchors_l[out_idx][k], shapes_l[out_idx][k], out_anchor, shape):
+                            is_ok = False
+                            break
+                    if not is_ok:
+                        continue
+                    break
+                break
+
+            if DEBUG_LVL >= 1:
+                print(f"""in_idx: {in_idx}, out_idx: {out_idx}, in_anchor: {
+                in_anchor}, in_shape: {shape}, out_anchor: {out_anchor}""")
+
+            in_idx_l[out_idx] = np.append(in_idx_l[out_idx], [in_idx], axis=0)
+            in_anchors_l[out_idx] = np.append(in_anchors_l[out_idx], [in_anchor], axis=0)
+            shapes_l[out_idx] = np.append(shapes_l[out_idx], [shape], axis=0)
+            out_anchors_l[out_idx] = np.append(out_anchors_l[out_idx], [out_anchor], axis=0)
+    return in_idx_l, in_anchors_l, shapes_l, out_anchors_l
+
+
+def numpyize(source_list):
+    if DEBUG_LVL >= 5:
+        print("Making this a data source: \n", source_list)
+    return lambda: source_list
+
+
+def get_pipeline(
+        batch_size=4,
+        in_size=None,
+        out_size=None,
+        even_paste_count=False,
+        k=4,
+        dtype=types.DALIDataType.UINT8,
+        no_intersections=True,
+        full_input=False,
+        in_anchor_top_left=False,
+        out_anchor_top_left=False
+):
+    pipe = Pipeline(batch_size=batch_size, num_threads=4, device_id=types.CPU_ONLY_DEVICE_ID)
+    with pipe:
+        input, _ = fn.file_reader(file_root=img_dir)
+        decoded = fn.image_decoder(input, device='cpu', output_type=types.RGB)
+        resized = fn.resize(decoded, resize_x=in_size[1], resize_y=in_size[0])
+        in_idx_l, in_anchors_l, shapes_l, out_anchors_l = prepare_cuts(
+            k, batch_size, in_size, out_size, even_paste_count,
+            no_intersections, full_input, in_anchor_top_left, out_anchor_top_left)
+        in_idx = fn.external_source(numpyize(in_idx_l))
```

Comment:
I don't get why the function is called numpyize. The input is already a numpy array. In my opinion `lambda: in_idx_l` would read better.

Reply:
It was left over from the time when it would do much more logic and the input was not in numpy array. Applying this